src/backend/api/meals.js

```diff
@@ -1,15 +1,130 @@
 const express = require("express");
+const app = require("../app");
+//const { max } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
+//Get meals that has a price smaller than maxPrice
 router.get("/", async (request, response) => {
+  console.log("akjdskank");
+  if ("maxPrice" in request.query) {
+    const dataTable = await knex("meal").where(
+      "price",
+      "<",
+      Number(request.query.maxPrice)
+    );
+    return response.json(dataTable);
+  }
+
+  //Get meals that still has available reservations
+  if ("availableReservation" in request.query) {
+    const availableReservations = await knex("meal")
+      .join("reservation", "reservation.meal_id", "=", "meal.id")
+      .where("max_reservations", ">", "number_of_guests");
+    response.json(availableReservations);
+  }
+  // Get meals that partially match a title
+  if ("title" in request.query) {
+    const getTitle = await knex("meal").where("title", "like", "%pasta%");
+    response.json(getTitle);
+  }
+  //Get meals that has been created after the date
+  // if (createdAfter in request.query) {
+  //   const getCreatedDate = await knex("meal").where(function () {
+  //     this.where("created_date", ">", "2022-01-20");
+  //   });
+  //   response.json(getCreatedDate);
+  // }
+  if ("createdAfter" in request.query) {
+    const createdAfter = await knex("meal").where(
+      "created_date",
+      ">",
+      "2022-01-20" //we can also give request.query.createdAfter
```

Comment:
yes, give that *request.query.createdAfter* . Don't hardcode the values because any other value (date in this case) won't work except what you provided